xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -2443,9 +2474,40 @@ public String getSkinFile(String filename, boolean forceSkinAction, XWikiContext
         return urlf.getURL(url, context);
     }
 
+    private Map<String, Object> getResourceURLCacheParameters(String resourceFilePath)
+    {
+        try {
+            URL resourceUrl = getResource(resourceFilePath);
+            return getResourceURLCacheParameters(resourceUrl);
+        } catch (MalformedURLException e) {
+            LOGGER.debug("Error while getting URL for resource path [{}]", resourceFilePath, e);
+            return Collections.singletonMap(CACHE_VERSION, getVersion());
+        }
+    }
+
+    private Map<String, Object> getResourceURLCacheParameters(URL resourceUrl)
+    {
+        Map<String, Object> parameters = new LinkedHashMap<>();
+
+        if (getURLConfiguration().useResourceLastModificationDate()) {
+            try {
+                Path resourcePath = Paths.get(resourceUrl.toURI());
+                FileTime lastModifiedTime = Files.getLastModifiedTime(resourcePath);
+                parameters.put(CACHE_VERSION, String.valueOf(lastModifiedTime.toMillis()));
+            } catch (Exception e) {
+                LOGGER.debug("Error when trying to access properties of resource URL [{}]", resourceUrl, e);
+                parameters.put(CACHE_VERSION, getVersion());
+            }
+        } else {
+            parameters.put(CACHE_VERSION, getVersion());
+        }
+
+        return parameters;
+    }
+
     public String getSkinFile(String filename, String skin, XWikiContext context)
     {
-        return getSkinFile(filename, skin, false, context);
+        return getSkinFile(filename, skin, context);
```

Comment:
Is this change on purpose ? Looks like an infinite loop.

Reply:
I don't think so. Too many changes and rollbacks on this one :)